File: sneaker-frontend/src/app/models/sneaker.ts
export class Sneaker {

  constructor(
    private _id: number,
    private _name: string,
    private _price: number,
    private _specialEdition: boolean,
    private _author: string,
    private _creationDate: Date,
    private _artistCollaborator: string
  ) {}

  public get artistCollaborator(): string {
    return this._artistCollaborator;
  }
  public set artistCollaborator(value: string) {
    this._artistCollaborator = value;
  }
  public get creationDate(): Date {
    return this._creationDate;
  }
  public set creationDate(value: Date) {
    this._creationDate = value;
  }
  public get author(): string {
    return this._author;
  }
  public set author(value: string) {
    this._author = value;
  }
  public get specialEdition(): boolean {
    return this._specialEdition;
  }
  public set specialEdition(value: boolean) {
    this._specialEdition = value;
  }
  public get price(): number {
    return this._price;
  }
  public set price(value: number) {
    this._price = value;
  }
  public get name(): string {
    return this._name;
  }
  public set name(value: string) {
    this._name = value;
  }
  public get id(): number {
    return this._id;
  }
  public set id(value: number) {
    this._id = value;
  }
}

File: sneakers/src/main/java/com/ironhack/sneakers/model/Sneaker.java
package com.ironhack.sneakers.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.util.Date;

@Entity
public class Sneaker {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String name;
    private double price;
    private boolean specialEdition;
    private String author;
    private Date creationDate;
    private String artistCollaborator;

    public Sneaker() {
    }

    public Sneaker(String name, double price, boolean specialEdition, String author, Date creationDate, String artistCollaborator) {
        this.name = name;
        this.price = price;
        this.specialEdition = specialEdition;
        this.author = author;
        this.creationDate = creationDate;
        this.artistCollaborator = artistCollaborator;
    }

    public Sneaker(Integer id, String name, double price, boolean specialEdition, String author, Date creationDate, String artistCollaborator) {
        this.id = id;
        this.name = name;
        this.price = price;
        this.specialEdition = specialEdition;
        this.author = author;
        this.creationDate = creationDate;
        this.artistCollaborator = artistCollaborator;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public boolean isSpecialEdition() {
        return specialEdition;
    }

    public void setSpecialEdition(boolean specialEdition) {
        this.specialEdition = specialEdition;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public Date getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(Date creationDate) {
        this.creationDate = creationDate;
    }

    public String getArtistCollaborator() {
        return artistCollaborator;
    }

    public void setArtistCollaborator(String artistCollaborator) {
        this.artistCollaborator = artistCollaborator;
    }
}
